gordon_janitor_gcp/http_client.py

```diff
@@ -169,3 +170,59 @@ def __init__(self, auth_client=None, session=None):
             json_callback = json.loads
         response = await self.request(method='get', url=url, **kwargs)
         return json_callback(response)
+
+
+class PagingGoogleClientMixin:
+    """HTTP client base class that extracts data from paginated Google APIs.
+    """
+
+    def parse_response_into_items(self, response, items):
+        """Parse an API response and append the results to ``items``
+
+        Must be implemented by inheriting class.
+
+        Args:
+            response (dict): API json response.
+            items (list): list for collecting parsed results.
+        """
+        raise NotImplementedError(
+            f'{self.__class__.__name__} implement "parse_response_into_items" '
+            'method!')
+
+    async def list_all_items(self, url, params, retries, retry_wait):
+        """Aggregate data from all pages of an API query.
+
+        Args:
+            url (str): Google API endpoint URL.
+            params (dict): URL query parameters.
+            retries (int): Number of times to retry an API call.
+            retry_wait (int): Number of seconds to wait between each retry.
+        Returns:
+            list: parsed query response results.
+        """
+        items = []
+        next_page_token = None
+
+        while True:
+            if next_page_token:
+                params['pageToken'] = next_page_token
+            try:
+                response = await self.get_json(url, params=params)
+            except exceptions.GCPHTTPError as e:
+                base_msg = f'Fetching items from "{url}" failed: {e}, '
+                if retries > 0:
```

Comment:
I think retry logic should actually live in the `AIOGoogleHTTPClient`, as well as allow retries to be configurable when instantiating the http client. It could live along side the refresh attempts (and maybe separated out into each of their own methods? if that could work?).

Reply:
I've decided to remove this altogether and leave it for another since it's not part of the ticket.